src/swapd/swap_state.rs

```diff
@@ -114,10 +124,287 @@ pub enum SwapCheckpointType {
     CheckpointAlicePreBuy,
 }
 
+pub enum Awaiting {
```

Comment:
I am a bit confused by the name `Awaiting`, can you comment on that, or describe here what the purpose of this enum is.